src/_pytest/config/__init__.py

```diff
@@ -357,6 +365,11 @@ def __init__(self) -> None:
         # If set, conftest loading is skipped.
         self._noconftest = False
 
+        # _getconftestmodules()'s call to _get_directory() causes a stat
+        # storm when it's called potentially thousands of times in a test
+        # session (#9478), often with the same path, so cache it.
+        self._get_directory = lru_cache(256)(_get_directory)
```

Comment:
Does it make sense to remove the bounds here, using `None` instead of 256?

Reply:
I just remembered I forgot to answer this.

I think it's always good to have limits on lru caches to avoid them growing too large. We can increase it though if it's too low for someone.